public/app/features/explore/Logs/Logs.tsx

```diff
@@ -427,6 +464,105 @@ class UnthemedLogs extends PureComponent<Props, State> {
 
   scrollToTopLogs = () => this.topLogsRef.current?.scrollIntoView();
 
+  prepareTableFrame = memoizeOne((frame: DataFrame): DataFrame => {
+    const { timeZone, splitOpen, range } = this.props;
+
+    const { timeIndex } = getTimeField(frame);
+    const sortedFrame = sortDataFrame(frame, timeIndex, this.state.logsSortOrder === LogsSortOrder.Descending);
+
+    const [frameWithOverrides] = applyFieldOverrides({
+      data: [sortedFrame],
+      timeZone,
+      theme: config.theme2,
+      replaceVariables: (v: string) => v,
+      fieldConfig: {
+        defaults: {
+          custom: {},
+        },
+        overrides: [],
+      },
+    });
+    // `getLinks` and `applyFieldOverrides` are taken from TableContainer.tsx
+    for (const field of frameWithOverrides.fields) {
+      field.getLinks = (config: ValueLinkConfig) => {
+        return getFieldLinksForExplore({
+          field,
+          rowIndex: config.valueRowIndex!,
+          splitOpenFn: splitOpen,
+          range: range,
+          dataFrame: sortedFrame!,
+        });
+      };
+      field.config = {
+        custom: {
+          filterable: true,
+          inspect: true,
+        },
+      };
+    }
+
+    return frameWithOverrides;
+  });
+
+  prepareTableData = async () => {
+    const { logsFrames } = this.props;
+    if (!logsFrames || !logsFrames.length) {
+      this.setState({ tableFrame: undefined });
+      return;
+    }
+    // TODO: This does not work with multiple logs queries for now, as we currently only support one logs frame.
+    let dataFrame = logsFrames[0];
+
+    const { timeIndex } = getTimeField(dataFrame);
+    dataFrame = sortDataFrame(dataFrame, timeIndex, this.state.logsSortOrder === LogsSortOrder.Descending);
+
+    // create extract JSON transformation for every field that is `json.RawMessage`
+    const transformations = dataFrame.fields
```

Comment:
feels like this could be somehow helped by `logsFrame.ts`, because in that one we do find the field that has the json-encoded-labels, though:
1. right now it does not return the field (because of reasons, ping me for more info 😁 )
2. in theory you may want to decode every json-encoded field, even fields that are not standard-logs-frame-fields, but just "extra" fields added to the frame... maybe.. maybe not.
WDYT?

Reply:
I added a TODO for now to come back to this in aa09b38129620cfe79a15cb8810383dd68f57cb6. Do you think that's fine for now?

Reply:
sure!

Reply:
btw. you could also create a github-issue that tracks this feature, and add checkboxes to the github-issue (just an idea)